controllers/release/release_adapter.go

```diff
@@ -341,6 +453,58 @@ func (a *Adapter) getReleaseStrategyFromRelease() (*v1alpha1.ReleaseStrategy, er
 	return a.getReleaseStrategy(releasePlanAdmission)
 }
 
+// getSnapshotEnvironmentBinding returns the SnapshotEnvironmentBinding associated with the Release being processed.
+// That association is defined by matching Environment and Application as seen in the ReleasePlanAdmission.
+// If the SnapshotEnvironmentBinding is not found or the Get operation fails, an error will be returned.
+func (a *Adapter) getSnapshotEnvironmentBinding(environment *appstudioshared.Environment,
+	releasePlanAdmission *v1alpha1.ReleasePlanAdmission) (*appstudioshared.ApplicationSnapshotEnvironmentBinding, error) {
+	bindingList := &appstudioshared.ApplicationSnapshotEnvironmentBindingList{}
+	opts := []client.ListOption{
+		client.InNamespace(environment.Namespace),
+		client.MatchingFields{"spec.environment": environment.Name},
+	}
+
+	err := a.client.List(a.context, bindingList, opts...)
```

Comment:
isn't the binding in the managed workspace?

Reply:
Totally. Changed.